### ISSUE TITLE
Integer Parsing Errors

### ISSUE THREAD

**Issue by christianbrugger** [stephenberry/glaze#365]
Thank you for the great library!

When parsing integers I noticed some silent errors:

Negative numbers are accepted for unsigned types
```
const auto res = glz::read_json<unsigned int>("-100");  // no error, res = 100
```

Overflow & Underflows are accepted
```
const auto res = glz::read_json<int8_t>("200");  // no error, res = -56
const auto res = glz::read_json<int8_t>("-200");  // no error, res = 56
```

Very large numbers are accepted (2**200, 61 digits):
```
const auto res = glz::read_json<uint64_t>("1606938044258990275541962092341162602522202993782792835301376");  
// no error , res = 0
```

I think all of these cases should return a syntax error.

**Comment by stephenberry** [comment]
@christianbrugger
Thanks for this feedback. Yeah, I agree that these should all be errors. I'll work on this soon.

**Comment by stephenberry** [comment]
I merged a change that addresses most of these (#366). Negative numbers are now rejected for unsigned integers. And, any integers using less than 64bits are checked for underflow and overflow.

I'm going to close this issue and open a new one regarding very large numbers that cannot fit within int64_t or uint64_t, as it may take me some time to figure out how to do this efficiently.

But, at least smaller integer types, where issues are more likely to be seen, are now handled.
